### PR TITLE
[FAB-18531] RFC for extend chaincode lifecycle API to query all appro…

### PR DESCRIPTION
**Pull request by huishen24**
Signed-off-by: Hui Shen hui.shen@oracle.com

This RFC is proposed to extend chaincode lifecycle api to query all approved chaincode definition

Thanks!

### CODE REVIEW

File: text/0000-fabric-extend-chaincode-lifecycle-api-to-query-all-aproved-chaincode [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+- Feature Name: extend_chaincode_lifecycle_function_to_query_all_approved_chaincode
+- Start Date: 2021-08-02
+- RFC PR: (leave this empty)
+- Fabric Component: fabric, fabric-protos
+- Fabric Issue: (leave this empty)
+
+# Summary
+[summary]: Extend chaincode lifecycle API to query all approved chaincodes
+
+Currently, Fabric peer CLI 'queryapproved' could query the details of an approved chaincode definitions, and this CLI requires a mandatory chaincode name parameter to query a specific chaincode definitions; however in some cases we expect to query out all approved chainocode definitions.
```
Comment (C0rWin) [inline text/0000-fabric-extend-chaincode-lifecycle-api-to-query-all-aproved-chaincode]:
```suggestion
Currently, Fabric peer CLI 'queryapproved' could query the details of an approved chaincode definitions, and this CLI requires a mandatory chaincode name parameter to query a specific chaincode definitions; however, in some cases we expect to query out all approved chaincode definitions.
```

File: text/0000-fabric-extend-chaincode-lifecycle-api-to-query-all-aproved-chaincode [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+- There is no new command is involved.
+- Command 'queryapproved' will be extended to support this function:
+  -- make parameter (n|-name <chaincode>) optional
+  -- if parameter (n|-name <chaincode>) is provided, keep current command behavior
+  -- if parameter (n|-name <chaincode>) is not provided, query the details of all approved chaincode, plus additional chaincode name in details output.
+
+* The example of query all approved chaincodes with JSON output format: *
+```
+$ peer lifecycle chaincode queryapproved --channelID <channelname> -o orderer.example.com:7050 --peerAddresses peer0.org1.example.com:7051 --output json
```
Comment (denyeart) [inline text/0000-fabric-extend-chaincode-lifecycle-api-to-query-all-aproved-chaincode]: No need for orderer interaction (-o).

**Review by denyeart — COMMENTED** [review]
This looks reasonable to me.
Note that the file should have .md suffix.

**Comment by satota2** [comment]
Hello @huishen24
I apologize for reaching out suddenly. I've noticed there hasn't been any recent updates. How is the current status of this matter?

I believe this RFC would be of great value to the community, and I would like to contribute to move it forward. Is there any way I can help? Alternatively, if you're no longer interested, would it be possible for me to take over the RFC?

**Comment by satota2** [comment]
Hello @huishen24 @denyeart @C0rWin

I pushed two additional commits to move this RFC forward. 
- To address the some review comments from @denyeart
- To update the explanation (remove unnnecesarry information, etc.)

I would appreciate it if you could take a moment to review these updates.

**Review by denyeart — APPROVED** [review]

**Comment by denyeart** [comment]
@hyperledger/fabric-core-maintainers I think this (small) RFC is ready for approval. Our policy requires 3 approvals for an RFC.

**Review by jt-nti — APPROVED** [review]

**Review by C0rWin — APPROVED** [review]

**Comment by denyeart** [comment]
3 approvals, I'll go ahead and merge.

**Comment by satota2** [comment]
@denyeart @C0rWin @jt-nti 
Thank you for the approvals and merge. I will start working on implementing this feature.

**Pull request merged**